tests/func/test_install.py

```diff
@@ -30,36 +30,11 @@ def test_should_create_hooks(self, git, dvc_repo):
             with open(self._hook(fname), "r") as fobj:
                 assert command in fobj.read()
 
-    def test_should_append_hooks_if_file_already_exists(self, git, dvc_repo):
+    def test_should_fail_if_file_already_exists(self, git, dvc_repo):
         with open(self._hook("post-checkout"), "w") as fobj:
-            fobj.write("#!/bin/sh\n" "echo hello\n")
+            fobj.write("hook content")
 
-        assert main(["install"]) == 0
-
-        expected_script = (
-            "#!/bin/sh\n"
-            "echo hello\n"
-            '[ "$3" = "0" ]'
-            ' || [ -z "$(git ls-files .dvc)" ]'
-            " || exec dvc checkout\n"
-        )
-
-        with open(self._hook("post-checkout"), "r") as fobj:
-            assert fobj.read() == expected_script
-
-    def test_should_be_idempotent(self, git, dvc_repo):
-        assert main(["install"]) == 0
-        assert main(["install"]) == 0
-
-        expected_script = (
-            "#!/bin/sh\n"
-            '[ "$3" = "0" ]'
-            ' || [ -z "$(git ls-files .dvc)" ]'
-            " || exec dvc checkout\n"
-        )
-
-        with open(self._hook("post-checkout"), "r") as fobj:
-            assert fobj.read() == expected_script
+        assert main(["install"]) != 0
```

Comment:
While we are at it, let's keep it more granular.
```suggestion
        with pytest.raises(GitHookAlreadyExistsError):
            dvc_repo.scm.install()
```